### ISSUE TITLE
Alert that shows changes in DHCP leases

### ISSUE THREAD

**Issue by mljdivemaster** [ZachChristensen28/pihole_dns_app#68]
I have several pi-hole installations in different locations, all connected to Splunk. I want to use the DHCP information to track changes when a new lease is added or removed. The purpose is to emulate like arpwatch.

Construct an alert using the lookup pihole_dhcp_lease_lookup for current values and compare them with known ones. 

The alert will be formatted in Slack messages like this:
![image](https://github.com/ZachChristensen28/pihole_dns_app/assets/19711259/86c6a489-025c-4cbc-9e09-e236816106b8)

I could use the raw data in the index but I rather use the lookup if we can get the dvc attribute in

**Comment by mljdivemaster** [comment]
I have this search as a start:

`|  inputlookup pihole_dhcp_lease_lookup
|  eval current="1"
|  append 
    [| inputlookup dhcpref
    |  search current!=missing
    |  fields - current]
|  eval known=case(isnull(current),"1")
|  stats first(current) as current, first(known) as known first(*) as * by dest_mac, dest_ip, dest_nt_host, dvc
|  fillnull value=new known
|  fillnull value=missing current
|  outputlookup dhcpref
|  eval change=case(
known=="new", known,
current=="missing", current)
|  search change=*
|  fields - known, current`

**Comment by mljdivemaster** [comment]
Need to create a lookup file and a lookup definition.

**Comment by mljdivemaster** [comment]
This is the result of the alert:
![image](https://github.com/ZachChristensen28/pihole_dns_app/assets/19711259/3945db53-6d07-4056-8be3-3ed4810f3785)

**Comment by mljdivemaster** [comment]
A Slack message can be formatted like this:

```
$result.change$  entry for mac $result.dest_mac$ and ip $result.dest_ip$ and host $result.dest_nt_host$ on DHCP server $result.dvc$
 
 $results_link$
```

**Comment by ZachChristensen28** [comment]
Another potential option is using streamstats. There is a good example in the [docs](https://docs.splunk.com/Documentation/Splunk/9.0.0/SearchReference/Streamstats#5._Calculate_when_a_DHCP_IP_lease_address_changed_for_a_specific_MAC_address) that looks interesting.

**Comment by mljdivemaster** [comment]
Yes, I have considered that. However, when a device is released and missing, you need to compare current data with a known state. So if a mac address is not covered in the timerange for streamstats, you won't be able to detect a missing device. Some DHCP clients are expected to disappear but for infrastructure devices, like cameras, you might want to get notified.

By using a lookup with a known state, the alert will trigger for missing devices, even if Splunk have been down for a while
